dotstrings/parser.py

```diff
@@ -14,7 +14,7 @@ class Patterns:
 
     comment = re.compile(r"(\'(?:[^\'\\]|\\[\s\S])*\')|//.*|/\*(?:[^*]|\*(?!/))*\*/", re.MULTILINE)
     whitespace = re.compile(r"\s*", re.MULTILINE)
-    entry = re.compile(r'"(.*)"\s*=\s*"(.*)";')
+    entry = re.compile(r'"([^"]*?)"\s*=\s*"((?:[^";]|"(?!\s*;))*?)";', re.DOTALL)
```

Comment:
Unfortunately this will break on strings that contain quotes. e.g. `NSLocalizedString("Hello \"World\"", "Some Comment")` gets turned into `"Hello \"World\"" = "Hello \"World\"";`. 

I think the second change on this line will also stop the string containing `;`.

Reply:
The first change is there so that the first match group doesn't end up matching the whole string (even matching the "value" part).

But yeah it wouldn't work when there are more than two quotes, in the "key" part of the string (the two quotes around the key itself). This was intentional because I didn't thought there would be any "key" that would have more quotes than two.

Reply:
The second change is finding a ";" to match the value part of the string.

Reply:
Can you change the test file I have here to contain edge cases?